type.go

```diff
@@ -36,11 +38,60 @@ func stringTypeOf(i interface{}) (string, error) {
 	}
 	ct := cassaType(i)
 	if ct == gocql.TypeCustom {
-		return "", fmt.Errorf("Unsupported type %T", i)
+		ct = gocql.TypeVarchar
+		// return "", fmt.Errorf("Unsupported type %T", i)
 	}
 	return cassaTypeToString(ct)
 }
 
+type structure struct {
+	A string
+        B int
+}
+
+func ProcessValue(i interface{}) (interface{}, error) {
+	if i == nil {
+		return nil, nil
+	}
+	_, isByteSlice := i.([]byte)
+	if !isByteSlice {
+		// Check if we found a higher kinded type
+		switch reflect.ValueOf(i).Kind() {
+		case reflect.Slice:
+			elemVal := reflect.Indirect(reflect.New(reflect.TypeOf(i).Elem())).Interface()
+			ct := cassaType(elemVal)
+			if ct == gocql.TypeCustom {
+				s := reflect.ValueOf(i)
+				result := make([]string, 0, s.Len())
+				for j := 0; j < s.Len(); j++ {
+					txt, err := json.Marshal(s.Index(j).Interface())
+					if err != nil {
+						return "", fmt.Errorf("Unsupported translation of value type %T", err)
+					}
+					result = append(result, string(txt))
+				}
+				return result, nil
+			}
+			return i, nil
+		case reflect.Map:
+			keyVal := reflect.Indirect(reflect.New(reflect.TypeOf(i).Key())).Interface()
```

Comment:
it seems this switch block is the same as the counterpart in stringTypeOf(), please check